lib/types.nix

```diff
@@ -818,34 +826,64 @@ rec {
 
     # A value from a set of allowed ones.
     enum = values:
+      #assert lib.assertMsg (isList values)
+      #  "enum: `values` must be a list: ${toPretty {} values}";
+      enumWith (map (v: nameValuePair v v) (lib.unique values));
```

Comment:
I see, `lib.unique` is probably particularly slow. I added it not to accidentaly trigger an assertion  (it is rather arbitrary, so I will remove it anyway)  in `enumWith`, which is off now because of some laziness tricks.

I think explicit interface assertion is much more usefull for debugging than a random exception in the implementation, so is there a way to keep it?